### PR TITLE
Implemented Comand events parsing #51

### PR DESCRIPTION
**Pull request by patriotyk**

### CODE REVIEW

File: src/comands.cc [hunk 1]
```diff
@@ -0,0 +1,167 @@
+#include "comands.h"
```
Comment (jerrytrieu) [inline src/comands.cc]: Typo, should be "commands.h"

File: src/comands.cc [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+namespace Comand {
```
Comment (jerrytrieu) [inline src/comands.cc]: Should be lowercase namespace `command`

File: src/comands.h [hunk 3]
```diff
@@ -0,0 +1,93 @@
+#ifndef COMANDS_H_
+#define COMANDS_H_
+#include <json/json.h>
+#include <string>
+#include "types.h"
+
+namespace Comand {
+
+class BaseComand {
```
Comment (jerrytrieu) [inline src/comands.h]: Should be `BaseCommand`

File: src/comands.h [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+ public:
```
Comment (jerrytrieu) [inline src/comands.h]: Can be struct if only public?

File: src/comands.h [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  std::string variant;
+  Json::Value toJson();
+};
+
+class Authenticate : public BaseComand {
+ public:
+  Authenticate(const data::ClientCredentials& client_credentials_in);
+  data::ClientCredentials client_credentials;
+  std::string toJson();
```
Comment (jerrytrieu) [inline src/comands.h]: Confusing to have same name as non-virtual function in base with different return type?

File: src/comands.h [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  static Authenticate fromJson(const std::string& json_str);
```
Comment (jerrytrieu) [inline src/comands.h]: Can be a templated function to reuse across types?

Comment (patriotyk) [inline src/comands.h]: Which function?

Comment (jerrytrieu) [inline src/comands.h]: It's ok, we can refactor later.

File: src/comands.h (hunk 5 above)
Comment (patriotyk) [inline src/comands.h]: why do You think it is confusing? In any way I call it through BaseCommand

Comment (jerrytrieu) [inline src/comands.h]: It's better to avoid this. Though the different return type will be caught at compile...? Refactor later.

**Pull request merged**